pkg/scheduler/scheduler.go

```diff
@@ -512,11 +512,13 @@ func (s *Scheduler) patchScheduleResultForResourceBinding(oldBinding *workv1alph
 		return nil
 	}
 
-	_, err = s.KarmadaClient.WorkV1alpha2().ResourceBindings(newBinding.Namespace).Patch(context.TODO(), newBinding.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{})
```

Comment:
Note: restart `karmada-controller-manager` will also increment the `generation` of RB, but it will not enter the scheduling process at this time.

Reply:
> restart karmada-controller-manager will also increment the generation of RB

Why?

Reply:
@chaunceyjiang is right. Through my analyze, it's mainly because of detector. For resource templates, we add it to the queue directly for adding events. However, for update events, we only add it to queue if spec of resource templates changes for performance. 

https://github.com/karmada-io/karmada/blob/ec7b3b14c66de00b43b11b517cad89baa8bbc382/pkg/detector/detector.go#L280-L308

After the status is aggregated into the resource template, the resourceVersion in rb.spec.resource is inconsistent with that in the actual resource template. It's currently working without issue. However, when controller-manager restarts, detector will add all resource templates again. So we will update rb.spec.resource.resourceVersion to the latest and then generation +1. But for scheduler, it think it do not need to schedule and not enter the scheduling process. So generation and schedulerObservedGeneration is not the same.

Reply:
In this case, return the latest binding after patching scheduler results will not work but the former redundancy design works.

Reply:
> but the former redundancy design works.

Yes.

Reply:
> However, when controller-manager restarts, detector will add all resource templates again. So we will update rb.spec.resource.resourceVersion to the latest and then generation +1. But for scheduler, it think it do not need to schedule and not enter the scheduling process. So generation and schedulerObservedGeneration is not the same.

Got it. Thanks.

Reply:
> After the status is aggregated into the resource template, the resourceVersion in rb.spec.resource is inconsistent with that in the actual resource template. It's currently working without issue. However, when controller-manager restarts, detector will add all resource templates again. So we will update rb.spec.resource.resourceVersion to the latest and then generation +1.


By the way, currently the graceful-eviction-controller relies too much on this mechanism. I think the graceful-eviction-controller should also handle create event. 

https://github.com/karmada-io/karmada/pull/3475


@RainbowMango @Poor12 @RainbowMango What do you think?

Reply:
> I think the graceful-eviction-controller should also handle create event.

What do you mean `handle create event`?

Reply:
Do we need to do something else during this PR?

Reply:
> handle create event.

https://github.com/karmada-io/karmada/blob/master/pkg/controllers/gracefuleviction/rb_graceful_eviction_controller.go#L87

Reply:
> Do we need to do something else during this PR?

No, This PR can solve my problem .